flax/linen/activation.py

```diff
@@ -67,8 +67,11 @@ def __call__(self, inputs: Array) -> Array:
     Returns:
       The transformed input.
     """
+    dtype = inputs.dtype
+    assert jnp.issubdtype(dtype, jnp.floating)
     negative_slope = self.param(
       'negative_slope',
-      lambda k: jnp.asarray(self.negative_slope_init, jnp.float32)
+      lambda k: jnp.asarray(self.negative_slope_init, dtype)
```

Comment:
Sure, I can do that.  Here, I was assuming the parameter always has type `float` since that's how it's annotated.  And then I'm casting it to the input dtype.  But now that I think about it again, I guess it does make sense to store the parameter in a custom dtype.

Reply:
Done.